apollo-router/src/plugins/record_replay/record.rs

```diff
@@ -0,0 +1,342 @@
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use futures::stream::once;
+use futures::StreamExt;
+use serde_json::json;
+use tokio::fs;
+use tower::BoxError;
+use tower::ServiceBuilder;
+use tower::ServiceExt as TowerServiceExt;
+
+use super::recording::RequestDetails;
+use super::recording::ResponseDetails;
+use super::recording::Subgraph;
+use super::recording::Subgraphs;
+use crate::context::Context;
+use crate::layers::ServiceBuilderExt;
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::services::execution;
+use crate::services::external::externalize_header_map;
+use crate::services::router;
+use crate::services::subgraph;
+use crate::services::supergraph;
+use crate::spec::query::Query;
+use crate::spec::Schema;
+use crate::Configuration;
+
+const RECORD_HEADER: &str = "x-apollo-router-record";
+const RECORD: &str = "record";
+const CLIENT_REQUEST: &str = "client_request";
+const CLIENT_RESPONSE: &str = "client_response";
+const QUERY_PLAN: &str = "query_plan";
+const SUBGRAPHS: &str = "subgraphs";
```

Comment:
Can we namespace these context keys by `experimental_record::` or something like this to avoid collision ?